detect_secrets/core/baseline.py

```diff
@@ -80,6 +99,9 @@ def initialize(
     for file in sorted(files_to_scan):
         output.scan_file(file)
 
+        if from_commit:
+            output.files_scanned.append(file)
```

Comment:
This line had me thinking for a while. Why do we need to keep track of all the files that we scan in the baseline?

By definition, this would produce a different set of files as compared to `git ls-files`. However, you could also argue that scanning the differential wouldn't seek to create a baseline as we know it, rather, attempts to see whether there's any secrets between the git commit provided, and current HEAD.

If this is true (and there's no other need for having a list of files scanned), you could just use the `results` field to see if there are any secrets between the `from_commit` and HEAD. If there isn't, our job is done. If there is, it's not like we'll be creating a baseline from it anyway?

Granted, this is overloading the baseline format as another way to present detected secrets, but :shrug:

Reply:
In this implementation it was necessary due to problems if we're passing the `--update` flag, as we need to know which files were 'scanned'; if a scan has found that secrets have been removed then the `new_results` won't have a key for eg. 'fileA', whereas `old_results` may do. In this case, we need to not include `old_results['fileA']` during our `merge_results` call.

I actually struggled with the changed files concept; if we proceed with this idea I'll revisit this to use `scan_diff` as it sounds like it solves the problem! My concern was that it's possible to change a file and not commit, scan, then revert the file. We'd now have secrets entries which have been invalidated.